engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/kor/KorRRNValidator.java

```diff
@@ -0,0 +1,29 @@
+
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.constraintvalidators.hv.kor;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+import java.util.regex.Pattern;
+
+import org.hibernate.validator.constraints.kor.KorRRN;
+
+public class KorRRNValidator implements ConstraintValidator<KorRRN, CharSequence> {
+
+	private static final Pattern KOR_RRN_REGEX = Pattern.compile(
+			"\\d{2}(0[1-9]|1[0-2])(0[1-9]|[12]\\d|3[01])-[1234]\\d{6}$" );
```

Comment:
Do you think it is worth to check the check digit for the older numbers?

> c, the 13th digit, is a check digit, used to verify that the number has been transcribed correctly. It is generated from the rest of the digits using the following algorithm (digits lettered a through m in order):
m = [11 − {(2a + 3b + 4c + 5d + 6e + 7f + 8g + 9h + 2i + 3j + 4k + 5l) mod 11}] mod 10

Reply:
What does "old number" mean?

Reply:
What is meaning "olrder number"?

Reply:
From what  I understand, for RRNs issued prior to October 2020 the last digit in the number was a check-digit. After that date, it became a random number. So I was asking if it would make sense to validate the chec-digit for numbers issued prior to October 2020.

Reply:
As far as I know, RRNs are still using check digits after October 2020.  Let's say "RRN" is "ABCDEF-GHIJKLM", then
Since October 2020, 'HIJKL' has been changed to a randomized order, but check digits are valid ('M')

[Configuration of RRNs](https://namu.wiki/w/주민등록번호#구성)
![Uploading Screenshot 2024-01-15 at 5.09.41 PM.png…]()

Reply:
I checked and the law is ambiguous. 

See [Government homepage 2020.05](https://www.law.go.kr/LSW/lsInfoP.do?lsiSeq=222265&ancYd=20201005&ancNo=00204&efYd=20201005&nwJoYnInfo=N&efGubun=Y&chrClsCd=010202&ancYnChk=0#0000)

See [Government homepage 2020.05](https://www.law.go.kr/LSW/lsInfoP.do?lsiSeq=222265&ancYd=20201005&ancNo=00204&efYd=20201005&nwJoYnInfo=N&efGubun=Y&chrClsCd=010202&ancYnChk=0#0000)

So, I'll check with the government tomorrow.

Reply:
Thank you! that would be nice.

Also as an idea: we could try adding an attribute to the constraint annotation:
```java
public @interface KorRRN {
	// ... 
	boolean validateCheckDigit() default true;
}
```
to make the validation optional. Or take it a step further and:
```java
public @interface KorRRN {
        //...
	ValidateCheckDigit validateCheckDigit() default ValidateCheckDigit.ALWAYS;
	
	enum ValidateCheckDigit {
		NEVER,
		ALWAYS,
		BEFORE_OCTOBER_2020_ONLY
	}
}
```

Reply:
I called the Korean goverment and they said you're right. 😂 This seems like a good idea. So I Updated KorRRN to include "validateCheckDigit"attribute and i added a few test cases.